### PR TITLE
fix: Set vnetCidr by the address space of the vnet

### LINKED ISSUE #2712: aks-engine does not set correct vnetCidr for agent pools
**Describe the bug**
Current aks-engine always sets vnetCidr for all agent pools to `10.0.0.0/8`.  From the code in https://github.com/Azure/aks-engine/blob/master/parts/k8s/kuberneteswindowssetup.ps1#L92, caller should set "vnetCidr" to the address space of the vnet used by cluser.

But vnetCidr is only set in https://github.com/Azure/aks-engine/blob/master/pkg/engine/params.go#L64 to the value of MasterProfile.VnetCidr.
```
			if properties.OrchestratorProfile.IsKubernetes() && properties.MasterProfile.VnetCidr != "" {
				addValue(parametersMap, "vnetCidr", properties.MasterProfile.VnetCidr)
			}
```

If caller sets MasterProfile.VnetCidr to the address space of the vnet used by cluster, below may be wrong.
https://github.com/Azure/aks-engine/blob/master/pkg/api/defaults.go#L581
```
		if p.OrchestratorProfile.OrchestratorType == Kubernetes {
			p.MasterProfile.FirstConsecutiveStaticIP = p.MasterProfile.GetFirstConsecutiveStaticIPAddress(p.MasterProfile.VnetCidr)
		}
```

Reference:
https://github.com/Azure/AKS/issues/1367
https://github.com/Azure/application-gateway-kubernetes-ingress/issues/535

**Steps To Reproduce**

Created a vnet (172.17.0.0/16) with a subnet (172.16.0.0/24) and then use this vnet to create a k8s cluster. RP sets vnectCidrs for Windows agent pool to ["172.17.0.0/16"] but ExceptionLists is still as below. 
```
"Policies":  [
                         {
                             "ExceptionList":  [
                                                   "10.0.0.0/8",
                                                   "172.17.0.0/24",
                                                   "10.240.0.0/16"
                                               ],
                             "Type":  "OutBoundNAT"
                         },
                         {
                             "DestinationPrefix":  "172.18.0.0/24",
                             "NeedEncap":  true,
                             "Type":  "ROUTE"
                         },
                         {
                             "Type":  "L2Driver"
                         }
                     ],
```

**Expected behavior**

```
"Policies":  [
                         {
                             "ExceptionList":  [
                                                   "172.17.0.0/16",
                                                   "10.240.0.0/16"
                                               ],
                             "Type":  "OutBoundNAT"
                         },
                         {
                             "DestinationPrefix":  "172.18.0.0/24",
                             "NeedEncap":  true,
                             "Type":  "ROUTE"
                         },
                         {
                             "Type":  "L2Driver"
                         }
                     ],
```

**AKS Engine version**
v-0.45.0-1

**Kubernetes version**
1.15.5

**Additional context**
I tested with adding below code in getParameters in pkg/engine/params.go and it works.
```
addValue(parametersMap, "vnetCidr", properties.AgentPoolProfiles[0].VnetCidrs[0])
```
Logs in `CustomDataSetupScript.log`
```
Filtering CNI config exception list values to work around WS2019 issue processing rules. Original exception list: 172.17.0.0/24 10.240.0.0/16 172.28.0.0/16, processed exception list: 172.17.0.0/16 10.240.0.0/16
```

Get-HnsEndpoint | ConvertTo-Json -Depth 10
```
        "Policies":  [
                         {
                             "ExceptionList":  [
                                                   "172.17.0.0/16",
                                                   "10.240.0.0/16"
                                               ],
                             "Type":  "OutBoundNAT"
                         },
                         {
                             "DestinationPrefix":  "172.18.0.0/24",
                             "NeedEncap":  true,
                             "Type":  "ROUTE"
                         },
                         {
                             "Type":  "L2Driver"
                         }
                     ],
```
```

### PR DESCRIPTION
**Pull request by AbelHu**
**Reason for Change**:

Fix #2712 

**Issue Fixed**:

Set vnetCidr by the address space of the vnet

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/engine/params.go [hunk 1]
```diff
@@ -78,6 +78,9 @@ func getParameters(cs *api.ContainerService, generatorCode string, aksEngineVers
 	}
 	if properties.HostedMasterProfile != nil {
 		addValue(parametersMap, "masterSubnet", properties.HostedMasterProfile.Subnet)
+		if len(properties.AgentPoolProfiles) > 0 && len(properties.AgentPoolProfiles[0].VnetCidrs) > 0 {
```
Comment (marosset) [inline pkg/engine/params.go]: Is it safe to hardcode the values from the first agent pool profile?
Does AKS only create one agent pool at a time (and will this continue to be the case)?

If it is safe can we add a comment here?

Comment (marosset) [reply pkg/engine/params.go]: @jackfrancis or @mboersma for another set of eyes. 
I'm not familiar with how aks-engine is used in AKS.

Comment (AbelHu) [reply pkg/engine/params.go]: AKS only supports to use one vnet. When a user creates a new k8s cluster, a custom vnet can be specified and the cidr of the agent pool will be set. If the custom vnet is not specified, DefaultVNETCIDR will be set. A cluster must have at least one agent pool so anyway the VnetCidrs of the default agent pool (index: 0) should be set.

**Comment by marosset** [comment]
Thanks @mboersma

**Pull request merged**